### PR TITLE
Move danger to CircleCI

### PR DESCRIPTION
**Pull request by ds300**
In this PR I've moved our Danger checks to CircleCI because travis was slowing us down.

Here's what was involved:

- Moved the Swift dangerfile logic into the TS dangerfile.
- Switched on the 'Only build PRs [and master]' setting in CircleCI because otherwise we could get false positive danger results.
- Created github actions to trigger the CircleCI builds for beta and promotion branches, since Circle doesn't give us a way to do that (feature request tracked [here](https://ideas.circleci.com/ideas/CCI-I-215))
- Cleaned up some duplicate keys in the changelog (the node yaml parser is stricter than the swift one)

✅ I have manually tested that danger is still working:

image

Including the changelog checks that were in the swift file:

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -184,7 +184,8 @@
     "tslint-plugin-relay": "0.0.2",
     "typescript": "3.8.3",
     "typescript-styled-plugin": "0.13.0",
-    "vscode-apollo-relay": "1.5.0"
+    "vscode-apollo-relay": "1.5.0",
+    "yaml": "^1.9.2"
```
Comment (ashfurrow) [inline package.json:188]: Pretty sure the `yaml` dependency is safe to include sans an RFC 😅 Especially since it's a dev dependency.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The CHANGELOG is not valid YML</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Missing Test Files:

- `src/lib/Components/Artist/__tests__/ArtistTestThing-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ds300** [comment]
@ashfurrow this is ready for a proper review now 🍏

File: dangerfile.ts [hunk 2]
```diff
@@ -56,36 +57,81 @@ If these files are supposed to not exist, please update your PR body to include
   callout(output)
 }
 
-if (fs.existsSync("tsc_raw.log")) {
-  const log = fs.readFileSync("tsc_raw.log")
-  if (log.length) {
-    fail("TypeScript hasn't passed, see below for full logs")
-    markdown(`### TypeScript Fails\n\n\`\`\`${log}\`\`\``)
+// Want to improve this file?
+// run `swift run danger-swift edit` in the terminal
```
Comment (ashfurrow) [inline dangerfile.ts:61]: 👍

File: .github/actions/promote.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# CircleCI doesn't automatically run builds on the app_store_submission branch (only master and PRs)
```
Comment (ashfurrow) [inline .github/actions/promote.yml:1]: Ah, nice workaround here 👍

File: dangerfile.ts (hunk 2 above)
Comment (ds300) [reply dangerfile.ts:61]: oh, uh, that comment probably should have been removed 🙈

Comment (ashfurrow) [reply dangerfile.ts:61]: Ahhhh right, I'll get it 😅

**Pull request merged**